client/src/modules/enterprises/enterprises.js

```diff
@@ -66,13 +69,19 @@ function EnterpriseController(Enterprises, Currencies, util, Notify, Projects, M
       return;
     }
 
+    // make sure only fresh data is sent to the server.
+    if (form.$pristine) {
+      Notify.warn('FORM.WARNINGS.NO_CHANGES');
+      return;
+    }
+
     var promise;
     var creation = (vm.hasEnterprise === false);
-    var enterprise = cleanEnterpriseForm(vm.enterprise);
+    var changes = util.filterFormElements(form, true);
```

Comment:
This seems much more maintainable than curating a hard coded object 👍 Good catch.